### PR TITLE
fix: Added Types in RecentLocalStorageItem Service

### PR DESCRIPTION
**Pull request by suyashpatil78**
### <samp>🤖 Generated by Copilot at cdff2ca</samp>

This pull request adds generic type parameters and type annotations to various method calls and services in the `fyle-mobile-app` project, to improve type safety and code readability. It also introduces a new interface `LoginInfo` to define the data structure for the login information stored in the local storage.

### <samp>🤖 Generated by Copilot at cdff2ca</samp>

> _Oh we're the coders of the sea, and we love our types so fine_
> _We use generics and interfaces, to make our code sublime_
> _We `get` and `set` with confidence, we know what we'll receive_
> _So heave away and pull the rope, on the count of three_

### <samp>🤖 Generated by Copilot at cdff2ca</samp>

*  Add `LoginInfo` interface to model the login information data stored in local storage ([link](https://github.com/fylein/fyle-mobile-app/pull/2073/files?diff=unified&w=0#diff-e92ae3158ca1ed3fa862e93037d61b371ee926dbb9f70e7785f4310714339b84R1-R5))
*  Use `LoginInfo` interface to type the data retrieved and stored by the `login-info.service.ts` methods ([link](https://github.com/fylein/fyle-mobile-app/pull/2073/files?diff=unified&w=0#diff-27b6db1c3724d634e7497c2be856170f652eb5035c0795bfda80b48ea445b106R6), [link](https://github.com/fylein/fyle-mobile-app/pull/2073/files?diff=unified&w=0#diff-27b6db1c3724d634e7497c2be856170f652eb5035c0795bfda80b48ea445b106L15-R16), [link](https://github.com/fylein/fyle-mobile-app/pull/2073/files?diff=unified&w=0#diff-27b6db1c3724d634e7497c2be856170f652eb5035c0795bfda80b48ea445b106L25-R26), [link](https://github.com/fylein/fyle-mobile-app/pull/2073/files?diff=unified&w=0#diff-27b6db1c3724d634e7497c2be856170f652eb5035c0795bfda80b48ea445b106L36-R39))
*  Add `RecentLocalStorageItems` interface to model the recent items data stored in local storage ([link](https://github.com/fylein/fyle-mobile-app/pull/2073/files?diff=unified&w=0#diff-8879b7c1094df38f9d5a2bc0eaa0a721229d608a5cce0dd2178807c518e3582dR5))
*  Add generic type parameters and type annotations to the `recent-local-storage-items.service.ts` methods to improve type checking and code completion ([link](https://github.com/fylein/fyle-mobile-app/pull/2073/files?diff=unified&w=0#diff-8879b7c1094df38f9d5a2bc0eaa0a721229d608a5cce0dd2178807c518e3582dL12-R15), [link](https://github.com/fylein/fyle-mobile-app/pull/2073/files?diff=unified&w=0#diff-8879b7c1094df38f9d5a2bc0eaa0a721229d608a5cce0dd2178807c518e3582dL30-R31), [link](https://github.com/fylein/fyle-mobile-app/pull/2073/files?diff=unified&w=0#diff-8879b7c1094df38f9d5a2bc0eaa0a721229d608a5cce0dd2178807c518e3582dL59-R60), [link](https://github.com/fylein/fyle-mobile-app/pull/2073/files?diff=unified&w=0#diff-8879b7c1094df38f9d5a2bc0eaa0a721229d608a5cce0dd2178807c518e3582dL69-R76))
*  Add generic type parameters to the `storageService.get` method calls in `report.service.ts`, `transaction.service.ts`, `add-edit-expense.page.ts`, `add-edit-mileage.page.ts`, and `add-edit-per-diem.page.ts` to specify the expected return types ([link](https://github.com/fylein/fyle-mobile-app/pull/2073/files?diff=unified&w=0#diff-12dc98777c493c39a3e7aec29b439cc7863ddfb002bf1e52052c8aee485c63c9L327-R327), [link](https://github.com/fylein/fyle-mobile-app/pull/2073/files?diff=unified&w=0#diff-712edde8a2bf233766bbc21a32a8166358eca7a00851e1a25af976cff8b780b5L456-R456), [link](https://github.com/fylein/fyle-mobile-app/pull/2073/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bL1035-R1035), [link](https://github.com/fylein/fyle-mobile-app/pull/2073/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bL2485-R2485), [link](https://github.com/fylein/fyle-mobile-app/pull/2073/files?diff=unified&w=0#diff-04c216da6200af711aa0f53bcd7e253086762b5d123b0729fac3b1862ec5fc3eL959-R959), [link](https://github.com/fylein/fyle-mobile-app/pull/2073/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL830-R830))
*  Add type assertions to the `storageService.get` method calls in `login-info.service.ts` and `storage.service.ts` to indicate the parsed values ([link](https://github.com/fylein/fyle-mobile-app/pull/2073/files?diff=unified&w=0#diff-27b6db1c3724d634e7497c2be856170f652eb5035c0795bfda80b48ea445b106L25-R26), [link](https://github.com/fylein/fyle-mobile-app/pull/2073/files?diff=unified&w=0#diff-d1336fa6fb2285c51b6cf8e0831ea088dd62c1ee72c82bbe76449ab5fb54a482L24-R24))
*  Format the code to wrap the `map` operator in a new line for readability in `login-info.service.ts` ([link](https://github.com/fylein/fyle-mobile-app/pull/2073/files?diff=unified&w=0#diff-27b6db1c3724d634e7497c2be856170f652eb5035c0795bfda80b48ea445b106L36-R39))

## Clickup
https://app.clickup.com/t/85zt699p4

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/core/models/login-info.model.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export interface LoginInfo {
```
Comment (yksolanki9) [inline src/app/core/models/login-info.model.ts:1]: Use Partial<LoginInfo> instead of setting all properties as null

**Review by yksolanki9 — CHANGES_REQUESTED** [review]
Return types for methods missing

File: src/app/core/models/login-info.model.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
-export interface LoginInfo {
-  version?: Date[];
-  lastLoggedInVersion?: string;
-  lastLoggedInTime?: string;
-}
+export type LoginInfo = Partial<{
```
Comment (yksolanki9) [inline src/app/core/models/login-info.model.ts:1]: Dont use partial inside the model, use it when assigning type to variables

**Review by yksolanki9 — CHANGES_REQUESTED** [review]
For methods not returning value, set return type to void. 
Return types not adderd in storage service

**Review by Dimple16 — APPROVED** [review]

**Review by yksolanki9 — APPROVED** [review]

**Pull request merged**